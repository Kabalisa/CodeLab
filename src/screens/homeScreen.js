import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity } from 'react-native';
import homeImage from '../../assets/homeImage.jpg';
import arrow from '../../assets/whiteArrow.png';

const HomeScreen = ({ navigation }) => {
  return (
    <View style={styles.containerStyle}>
      <View style={styles.headerStyle}>
        <Text style={styles.headerTextStyle}>CodeLab</Text>
      </View>
      <View style={styles.secondContainerStyle}>
        <Text style={styles.textStyle}>Find Java Developers In Lagos</Text>
        <TouchableOpacity
          style={styles.highlightStyle}
          onPress={() => navigation.navigate('Login')}
        >
          <Image source={arrow} style={styles.arrowStyle} />
        </TouchableOpacity>
        <View style={styles.viewStyle}>
          <Image source={homeImage} style={styles.imageStyle} />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    height: '100%'
  },
  headerStyle: {
    height: '11%',
    flexDirection: 'row',
    borderBottomColor: '#5075D4',
    borderBottomWidth: 1
  },
  headerTextStyle: {
    color: '#5075D4',
    fontSize: 24,
    fontWeight: 'bold',
    alignSelf: 'flex-end',
    bottom: '2%',
    left: '10%'
  },
  viewStyle: {
    height: '100%',
    position: 'absolute'
  },
  secondContainerStyle: {
    height: '89%'
  },
  imageStyle: {
    height: '100%',
    opacity: 0.7
  },
  textStyle: {
    color: 'white',
    fontSize: 24,
    fontWeight: 'bold',
    marginHorizontal: '9%',
    top: '10%',
    zIndex: 2
  },
  arrowStyle: {
    width: 35,
    height: 35,
    zIndex: 2
  },
  highlightStyle: {
    marginHorizontal: '43%',
    top: '73%',
    zIndex: 2
  }
});

export default HomeScreen;
