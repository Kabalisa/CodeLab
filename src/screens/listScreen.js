import React, { Component } from 'react';
import { Text, View, StyleSheet, FlatList } from 'react-native';
import Header from '../components/header';
import UserCard from '../components/userCard';
import Footer from '../components/footer';

class ListScreen extends Component {
  render() {
    const { navigation } = this.props;
    const numbers = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'];
    return (
      <View>
        <Header />
        <View style={styles.viewStyle}>
          <Text style={styles.textStyle}>Java Developers In Lagos:</Text>
          <View style={styles.viewCardStyle}>
            <FlatList
              keyExtractor={one => one}
              data={numbers}
              renderItem={() => <UserCard navigation={navigation} />}
            />
          </View>
        </View>
        <Footer />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  textStyle: {
    fontSize: 18,
    fontWeight: 'bold',
    left: 15,
    top: 20
  },
  viewStyle: {
    height: 695,
    flexDirection: 'column'
  },
  viewCardStyle: {
    marginHorizontal: 30,
    marginTop: 50,
    height: 600
  }
});

export default ListScreen;
