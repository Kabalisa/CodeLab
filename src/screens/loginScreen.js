import React, { Component } from 'react';
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity,
  AsyncStorage
} from 'react-native';
import firebase from 'firebase';
import Footer from '../components/footer';
import icon from '../../assets/github-512.png';
import initializeFirebase from '../helpers/firebaseConfig';
import getGithubToken from '../helpers/getGithubToken';

const GithubStorageKey = 'githubtoken';
export const signIn = async token => {
  try {
    if (!token) {
      const token = await getGithubToken();
      if (token) {
        await AsyncStorage.setItem(GithubStorageKey, token);
        return signIn(token);
      } else {
        return;
      }
    }
    const credential = firebase.auth.GithubAuthProvider.credential(token);
    return firebase.auth().signInWithCredential(credential);
  } catch ({ message }) {
    alert(message);
  }
};

export const attemptToRestoreAuth = async () => {
  let token = await AsyncStorage.getItem(GithubStorageKey);
  if (token) {
    return signIn(token);
  }
};

class LoginScreen extends Component {
  state = { isSignedIn: false };

  componentDidMount() {
    this.setupFirebase();
  }

  setupFirebase = async () => {
    const { navigation } = this.props;
    initializeFirebase();

    firebase.auth().onAuthStateChanged(async auth => {
      const isSignedIn = !!auth;
      this.setState({ isSignedIn });
      isSignedIn && navigation.navigate('List');
      if (!isSignedIn) {
        attemptToRestoreAuth();
      }
    });
  };
  render() {
    return (
      <View style={styles.containerStyle}>
        <View style={styles.headerStyle}>
          <Text style={styles.headerTextStyle}>CodeLab</Text>
        </View>
        <View style={styles.parentStyle}>
          <TouchableOpacity onPress={() => signIn()}>
            <View style={styles.viewStyle}>
              <Text style={styles.textstyle}>Login With</Text>
              <Image source={icon} style={styles.imageStyle} />
            </View>
          </TouchableOpacity>
        </View>
        <Footer />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  containerStyle: {
    height: '100%'
  },
  headerStyle: {
    height: '11%',
    flexDirection: 'row',
    borderBottomColor: '#5075D4',
    borderBottomWidth: 1
  },
  headerTextStyle: {
    color: '#5075D4',
    fontSize: 24,
    fontWeight: 'bold',
    alignSelf: 'flex-end',
    bottom: '2%',
    left: '10%'
  },
  parentStyle: {
    flexDirection: 'column',
    alignItems: 'center',
    height: '83%',
    justifyContent: 'center'
  },
  viewStyle: {
    borderWidth: 1,
    borderColor: '#5075D4',
    padding: 5,
    flexDirection: 'row',
    justifyContent: 'center',
    width: 250,
    backgroundColor: '#5075D4',
    borderRadius: 8
  },
  imageStyle: {
    width: 50,
    height: 50
  },
  textstyle: {
    top: 16,
    right: 3,
    fontSize: 18,
    fontWeight: 'bold',
    color: 'white'
  }
});

export default LoginScreen;
