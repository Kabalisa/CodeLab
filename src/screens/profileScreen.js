import React, { Component } from 'react';
import {
  Text,
  View,
  Image,
  StyleSheet,
  Share,
  TouchableOpacity
} from 'react-native';
import Header from '../components/header';
import Footer from '../components/footer';
import userAvatar from '../../assets/userAvatar.png';
class ProfileScreen extends Component {
  handleShare = async username => {
    const githubUrl = `https://github.com/${username}`;
    try {
      await Share.share({
        message: `Check out this awesome developer @${username}, ${githubUrl}`,
        title: 'Lagos developer'
      });
    } catch (error) {
      alert(error.message);
    }
  };
  render() {
    const { navigation } = this.props;
    const item = navigation.getParam('item');
    return (
      <View>
        <Header navigation={navigation} />
        <View style={styles.viewStyle}>
          <Image
            source={{ uri: item.node.avatarUrl }}
            style={styles.imageStyle}
          />
          <Text style={styles.usernameStyle}>{item.node.login}</Text>
          <Text style={styles.textOneStyle}>Github URL:</Text>
          <Text
            style={styles.textTwoStyle}
          >{`https://github.com/${item.node.login}`}</Text>
          <View style={styles.viewOneStyle}>
            <Text style={styles.textThreeStyle}>
              {item.node.repositories.totalCount}
            </Text>
            <Text style={styles.textFourStyle}>repos</Text>
          </View>
          <View style={styles.viewTwoStyle}>
            <Text style={styles.textThreeStyle}>
              {item.node.starredRepositories.totalCount}
            </Text>
            <Text style={styles.textFourStyle}>starring</Text>
          </View>
          <TouchableOpacity onPress={() => this.handleShare(item.node.login)}>
            <View style={styles.buttonStyle}>
              <Text style={styles.textstyle}>Share Profile</Text>
            </View>
          </TouchableOpacity>
        </View>
        <Footer />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  viewStyle: {
    flexDirection: 'column',
    alignItems: 'center',
    height: 695
  },
  imageStyle: {
    width: 220,
    height: 220,
    borderRadius: 110,
    marginVertical: 15
  },
  usernameStyle: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 25
  },
  textOneStyle: { fontSize: 18 },
  textTwoStyle: { fontSize: 18, color: '#5075D4', marginBottom: 20 },
  viewOneStyle: { flexDirection: 'row', marginBottom: 20 },
  viewTwoStyle: { flexDirection: 'row', marginLeft: 15 },
  textThreeStyle: {
    fontSize: 24,
    fontWeight: 'bold',
    bottom: 6,
    marginRight: 10
  },
  textFourStyle: { fontSize: 18 },
  buttonStyle: {
    borderWidth: 1,
    borderColor: '#5075D4',
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'center',
    width: 230,
    backgroundColor: '#5075D4',
    borderRadius: 8,
    marginTop: 40
  },
  textstyle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: 'white'
  }
});

export default ProfileScreen;
