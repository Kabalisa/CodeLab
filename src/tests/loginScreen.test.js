import LoginScreen from '../screens/loginScreen';
import { signIn, attemptToRestoreAuth } from '../screens/loginScreen';
describe('LoginScreen tests', () => {
  const props = { navigation: { navigate: jest.fn() } };
  it('should render without error', () => {
    const wrapper = shallow(<LoginScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('call the signin method', () => {
    const result = signIn();
    expect(result).not.toBe(null);
  });
  it('call the signin method', () => {
    const result = signIn('token');
    expect(result).not.toBe(null);
  });
  it('call the attemptToRestoreAuth method', () => {
    const result = attemptToRestoreAuth();
    expect(result).not.toBe(null);
  });
  it('onPress test', () => {
    const wrapper = shallow(<LoginScreen {...props} />);
    const button = wrapper.find('TouchableOpacity');
    button.props().onPress();
  });
});
