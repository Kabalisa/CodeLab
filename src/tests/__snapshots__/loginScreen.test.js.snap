// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen tests should render without error 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#5075D4",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": "11%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "bottom": "2%",
          "color": "#5075D4",
          "fontSize": 24,
          "fontWeight": "bold",
          "left": "10%",
        }
      }
    >
      CodeLab
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "83%",
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#5075D4",
            "borderColor": "#5075D4",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 5,
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "right": 3,
              "top": 16,
            }
          }
        >
          Login With
        </Text>
        <Image
          source={1}
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </View>
    </TouchableOpacity>
  </View>
  <Footer />
</View>
`;
